### PR TITLE
Jjjoness/asset browser browsing assets fast in tree view results in freeze of editor 9739

### LINKED ISSUE #9739: Asset Browser: Browsing assets fast when Tree View is enabled results in constant freeze of Editor
**Describe the bug**
Browsing assets in the **Asset Browser** with **Tree View** option enabled results in the constant freeze of the Editor.
The **List View** is set as default view in the **Asset Browser**, and the freeze does **not** occur with it enabled.
To enable the **Tree View** click on the hamburger menu and click on **Tree View** in the dropdown menu.
Unfortunately, no logs are created when this issue occurs.

Please note that this issue also occurs on the Linux OS.

Similar Linux specific closed issue - https://github.com/o3de/o3de/issues/4659

**Steps to reproduce**
1. Type "a" in the **Asset Browser** search bar.
2. LMB on the slider and hover the slider to the bottom of the search results.
3. (Alternatively use scroll on mouse, it is easier to reproduce the issue by using the slider)

**Expected behavior**
The Editor remains stable at all times, browsing assets fast in the **Asset Browser** with **Tree View** enabled does **not** result in constant freeze.

**Actual behavior**
Browsing the assets fast in the **Asset Browser** with **Tree View** enabled results in constant freeze of the Editor.

**Video**

https://user-images.githubusercontent.com/86951967/170030139-9c631123-9b0d-45c9-914e-241f1919babc.mp4

**Found in Branch**
Development (840f88f)

**Desktop:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel Core i7-8700K
 - GPU: NVIDIA GeForce RTX 2080 Ti
 - Memory: 16GB

### PR DESCRIPTION
**Pull request by jjjoness**
Fixed #9739 by increasing the size of the Thread pool.

### CODE REVIEW

**Review by alexmontAmazon — APPROVED** [review]
looks good!

**Review by amzn-tmryan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.cpp [hunk 1]
```diff
@@ -25,7 +25,13 @@ namespace AzToolsFramework
         ThumbnailerComponent::ThumbnailerComponent()
             : m_missingThumbnail(new MissingThumbnail())
             , m_loadingThumbnail(new LoadingThumbnail())
+            , m_numLoading(0)
         {
+            // Increase the maximum number of QThreads to support the number of
+            // concurrent jobs needed here.
+            int currentMaxThreadCount = QThreadPool::globalInstance()->maxThreadCount();
+            int newMaxThreadCount = qMax<int>(currentMaxThreadCount, m_maxThumbnailJobs + 1);
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.cpp:33]: Nit: Prefer AZStd::max

File: Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.h [hunk 2]
```diff
@@ -68,6 +68,10 @@ namespace AzToolsFramework
             SharedThumbnail m_missingThumbnail;
             //! Default loading thumbnail used when thumbnail is found by is not yet generated
             SharedThumbnail m_loadingThumbnail;
+            //! Maximum number of concurrent jobs allowed.
+            const int m_maxThumbnailJobs{ 64 };
+            //! Current number of jobs running.
+            int m_numLoading;
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.h:74]: nit: a more descriptive name would help? (m_currentJobsCount?)

**Review by AMZN-daimini — APPROVED** [review]
Very minor notes, but looks good to me!

File: Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.cpp (hunk 1 above)
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.cpp:33]: No problem.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.h (hunk 2 above)
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Thumbnails/ThumbnailerComponent.h:74]: I'll change both before merging.

**Pull request merged**